mdns.go

```diff
@@ -126,9 +132,15 @@ func (c *mdnsConn) SendQuery(q *Query) error {
 }
 
 // SendResponse sends a response.
-// The message is sent as unicast, if an receiver address is specified in the response.
+// The message is:
+// 1. sent as legacy unicast, if destination port is not 5353,
+// 2. sent as unicast, if the receiver address is specified in the response,
 // Otherwise the message is sent multicast.
 func (c *mdnsConn) SendResponse(resp *Response) error {
+	if isLegacyUnicastSource(resp.addr) {
+		return c.sendResponseTo(resp.msg, resp.iface, resp.addr)
+	}
+
```

Comment:
Is the check for legacy unicast source actually needed here because the response is always sent as unicast if `resp.addr != nil`?

Reply:
A fair point there, I was mass-adding the checks across all the places in the code and didn't realize the code doesn't need to be here. This will get correctly handled in the `c.sendResponseTo(...)` call.

Reply:
Fixed in the new commit.